morituri/command/cd.py

```diff
@@ -134,10 +134,26 @@ def do(self):
         if not self.program.metadata:
             # fall back to FreeDB for lookup
             cddbid = self.ittoc.getCDDBValues()
-            cddbmd = self.program.getCDDB(cddbid)
-            if cddbmd:
-                sys.stdout.write('FreeDB identifies disc as %s\n' % cddbmd)
 
+            allow_freedb = getattr(self.options, 'allow_freedb', False)
+
+            sfdb = simplefreedb.SimpleFreeDB()
+            matches = sfdb.query(cddbid[0], cddbid[1], cddbid[2:-1], cddbid[-1])
```

Comment:
Maybe the cddbid parsing logic should be done inside of sfdb.query(), because this is pretty opaque. If you know the sfdb.query() signature, sure you can work it out, but outside of your test cases it doesn't seem like you're ever calling sfdb.query() without having an already composed cddbid.

Reply:
I'd rather have `getCDDBValues()` not build a custom array of stuff. It would indeed be simpler here to just pass `cddbid`, but the way `result` is crafted in `getCDDBValues()` makes me believe it was originally written by someone who had no clue about data structures. If you consider `SimpleFreeDB` could live outside the `whipper` project, it is saner to have this prototype, because the parameters are constructed separately: there is no such thing as "grab all the info and pack them in an array that way" thing outside of `whipper`.